rmw_fastrtps_cpp/src/rmw_wait.cpp

```diff
@@ -173,14 +172,6 @@ rmw_wait(
   // after we check, it will be caught on the next call to this function).
   lock.unlock();
 
-  // Even if this was a non-blocking wait, signal a timeout if there's no data.
-  // This makes the return behavior consistent with rcl expectations for zero timeout value.
-  // Do this before detaching the listeners because the data gets cleared for guard conditions.
-  bool hasData = check_wait_set_for_data(subscriptions, guard_conditions, services, clients);
-  if (!hasData && wait_timeout && wait_timeout->sec == 0 && wait_timeout->nsec == 0) {
-    timeout = true;
```

Comment:
Removing this is what causes the tests to fail. If the timeout is reached (even if the timeout is 0) [RMW_RET_TIMEOUT](https://github.com/ros2/rmw/blob/e8799d79e0bc1648cf58be5b9edf2b523be51585/rmw/include/rmw/types.h#L34) should be returned (or at least that's what the test expects)